### PR TITLE
feat: #52114 - added useEffectAsync hook as a reusable hook for async effect handling

### PR DESCRIPTION
**Pull request by Tobzor**
- Added the useEffectAsync hook which allows for an async function as input for the effect. 
  * It also provides an AbortSignal as input to the provided effect which can be used for cancelling http requests in progress
  * signal.aborted will be true whenever the component the hook is declared in is unmounted.

### CODE REVIEW

**Comment by CsabaTheKiss** [comment]
We have restrictions to the PR name, as described in the build error:
image

Could you add the related task ID as well?
In a format like:
feat: #123456 - Some relevant PR title.

File: src/hooks/useEffectAsync.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *       if (!signal.aborted) {
+ *           setState("some value")
+ *       }
+ *
+ *       return () => {
+ *           optionalCleanup();
+ *       }
+ * }, [])
+ */
+export function useEffectAsync(effect: AsyncEffect, deps: React.DependencyList = []): void {
```
Comment (CsabaTheKiss) [inline src/hooks/useEffectAsync.ts:21]: Could you add unit test as well?
We use `@testing-library/react-hooks` for test hooks without components.
An example on that:
```ts
import { act, renderHook } from '@testing-library/react-hooks';
...
        it('should return with the instCode description when the plant exists', () => {
            // given
            const instCode = 'JCA';
            const { result } = renderHook(() => useInstCodeDescription(instCode));

            // when
            expect(result.current).toBe('mock JCA description');
        });

```
More info about usage can be found here:
https://www.npmjs.com/package/@testing-library/react-hooks

File: src/hooks/useEffectAsync.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, { useEffect } from 'react';
+
+type AsyncEffect = (signal: AbortSignal) => Promise<void | (() => void)> | void;
+
+/**
```
Comment (CsabaTheKiss) [inline src/hooks/useEffectAsync.ts:5]: Could you add proper JS doc format param description as well here?
```ts
/**
 * 
 * @param effect 
 * @param deps 
 */
```

**Comment by CsabaTheKiss** [comment]
I think we need to do a version bump as well here, if we want to release a new version with this new hook. (0.2.7 => 0.2.8)

File: src/hooks/useEffectAsync.ts (hunk 2 above)
Comment (Tobzor) [reply src/hooks/useEffectAsync.ts:5]: Used the official React jsdoc for useeffect and added the description of the optional abortsignal usage. Good enough?

File: src/hooks/useEffectAsync.ts (hunk 1 above)
Comment (CsabaTheKiss) [reply src/hooks/useEffectAsync.ts:11]: As discussed: I'll add unit test for this later, just to not hold up the PR for long.

File: src/hooks/useEffectAsync.ts (hunk 2 above)
Comment (CsabaTheKiss) [reply src/hooks/useEffectAsync.ts]: Perfect, thanks! 👍

File: src/hooks/useEffectAsync.ts (hunk 1 above)
Comment (hkot1991) [reply src/hooks/useEffectAsync.ts:11]: Poor excuse for not writing tests, later === never in almost all cases

Comment (CsabaTheKiss) [reply src/hooks/useEffectAsync.ts:11]: Probably some context is missing here: I'm also have a task in echoUtils at the moment, where I need to add tests.
My plan was to add this test there.

But ofc you're right on being more strict about this: I'll add the test to this PR then.

Comment (CsabaTheKiss) [reply src/hooks/useEffectAsync.ts:11]: @hkot1991 added tests :)

**Review by CsabaTheKiss — APPROVED** [review]

**Comment by CsabaTheKiss** [comment]
One more thing: don't forget to export it through the hook/index.ts!

**Pull request merged**